### ISSUE TITLE
[Bug]: Please wrap private class polyfill into pure annotated IIFE so unused classes can get eliminated by minifiers

### ISSUE THREAD

**Issue by danieltroger** [babel/babel#13193]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Other (Next.js, Gatsby, vue-cli, ...)

### Input code

Before fix: https://babeljs.io/repl/#?browsers=last%201%20firefox%20version&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=MYGwhgzhAEC2CeEwFdgFNoG8BQ1oGIAHAJwEsA3MAFzUuOgF5oAiJVNZgbm1wJIupo26ABQBKLLzzE0VZMQB20KgAtSEAHREylGnW54AvjzwBzCkJSiJOPNNnylq9Vv67L7cQejHsv4AD2ChABIGgaIAGmIgpoAO5wiFZo4mLcQA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.13.15&externalPlugins=%40babel%2Fplugin-syntax-class-properties%407.12.13

After the fix that I ask you to do: https://babeljs.io/repl/#?browsers=last%201%20firefox%20version&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=G4QwTgBAtgngziArgYwKYQLwQPQCoACA-oQAoCqASgKLG7YAU9AlJgHwTIA2Icc08SNBADeAKAgQAxAAcwAS1AAXVKEhYARAhSp1AbnFTZCkMq1pmIgxLCpFiMADsIigBZy4AOhnylK8PokAXwMAcwVUM1QLMQlrW3snV3cvI19I5gCIYMCmDNFsbGQAewc4Is5UD04ikPoHVAB3fnSmJn0gA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.13.15&externalPlugins=%40babel%2Fplugin-syntax-class-properties%407.12.13

As you can see, after the fix this `new WeakMap(), _privatesauce = new WeakSet(),` is inside of a function so a minifier, i.e. terser, can remove it. Otherwise output code can look like this `!function(){new WeakMap,new WeakSet;!async function(){console.log("hi")}()}();`

### Current and expected behavior

Current behavior: babel puts

```
var _privatevar = new WeakMap();

var _privatesauce = new WeakSet();
```
into the outmost scope so it's left in minified output.

Expected behaviour: terser optimizes everything away

Just check the parcel issue (https://github.com/parcel-bundler/parcel/issues/6116) please if you don't understand

### Environment

parcel 2, please see the much higher effort issue with .zip to reproduce that I opened here: https://github.com/parcel-bundler/parcel/issues/6116

### Possible solution

Wrap into pure annotated IIFE like the title suggests

### Additional context

https://github.com/parcel-bundler/parcel/issues/6116

**Comment by nicolo-ribaudo** [comment]
I'm trying to think about where to exactly add the pure annotation.

How do you suggest to compile this code?
```js
class A {
  static #x = 1;
  static #y = console.log(this.#x);
  #z;
}
```

**Comment by danieltroger** [comment]
@kzc came up with the idea but if I understood it correctly it should look like this:

```
const A = (() => class A {
  static #x = 1;
  static #y = console.log(this.#x);
  #z;
})()
```

And then get transpiled like this: https://babeljs.io/repl/#?browsers=last%201%20firefox%20version&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=MYewdgzgLgBAgjAvDAFCglEgfDYAbAQwgnhgG8AoGGaAqAS2BgGIAPJGARgG4qao6jFgE8OoSCDwBTAHR4QAcxRQAFvQgy26XtWYAvXgF90GIA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Ctypescript&prettier=false&targets=&version=7.13.15&externalPlugins=%40babel%2Fplugin-syntax-class-properties%407.12.13

**Comment by nicolo-ribaudo** [comment]
I realized that I can add the pure annotation right before `new WeakMap`, and terser will remove it. Wrapping in a pure IIFE doesn't work, because that code would stop logging 1 when the class is removed by a minifier.

**Comment by danieltroger** [comment]
> Wrapping in a pure IIFE doesn't work, because that code would stop logging 1 when the class is removed by a minifier.

Ah shoot I didn't evaluate it enough in my head. It's great that I'm not the one making changes xD

> I realized that I can add the pure annotation right before new WeakMap, and terser will remove it.

That sounds like an ideal solution, very cool, thanks a lot!

(TWO HUNDRED AND TWENTY FOUR FILES CHANGED WHAAAAAAT, you're fast)

Btw I am once again asking for your support: how can I get private class properties and members just converted to public ones?
All these transpilations break terser's property mangling and make the output size bigger and I don't really care at this point. Is it possible to just rename private class properties to random public ones?

I still want to code with them private so I can ship them untranspiled once safari and firefox support them.

**Comment by nicolo-ribaudo** [comment]
No, currently Babel either compiles all the class fields (both public and private) or none of them.
In the future it will probably be possible to compile private fields only, but it's not possible yet.

**Comment by nicolo-ribaudo** [comment]
Fixed in 7.14.0!
